src/DataTable.php

```diff
@@ -147,6 +169,35 @@ protected function initColumns()
 
     }
 
+    /**
+     * Detect a model class from `dataProvider` or `data` attributes
+     *
+     * @see \yii\grid\DataColumn::getHeaderCellLabel()
+     *
+     * return Model|null    NULL is returned when only property $data is defined, and is either empty or first entry is not of type model
+     */
+    protected function getModel()
+    {
+        $provider = $this->_dataProvider;
+        if ($provider instanceof ActiveDataProvider && $provider->query instanceof ActiveQueryInterface) {
+            /* @var $modelClass Model */
+            $modelClass = $provider->query->modelClass;
```

Comment:
Does `ActiveDataProvider` not have own `modelClass` property?

Reply:
I took the most of this code from `\yii\grid\DataColumn::getHeaderCellLabel` so I think there is a reason for this checks.